### PR TITLE
Place a figure on a board

### PR DESCRIPTION
**Pull request by albekruz**

### CODE REVIEW

File: services/app/src/pages/game/Game.jsx [hunk 1]
```diff
@@ -16,8 +17,10 @@ function Room() {
         <button type="button">Copy link</button>
       </CopyToClipboard>
       <Player name={username} />
-      <ChessBoard numberOfColumns={6} numberOfRows={6} />
-      <ChessBoardPieces />
+      <ChessGameProvider>
+        <ChessBoard numberOfColumns={6} numberOfRows={6} />
```
Comment (jvalatkevicius) [inline services/app/src/pages/game/Game.jsx:21]: numberOfColumns ir numberOfRows yra aprasyti constantu faile `services/app/src/constants/board.js`

File: services/app/src/components/chessBoard/ChessBoard.jsx [hunk 2]
```diff
@@ -1,54 +1,41 @@
...
-    })),
-  );
+const ChessBoard = () => {
+  const { placeItemOnBoard, board } = useContext(ChessBoardContext)
+
+  const displayBoard = (
+     board.map((row, rowIndex) => (
+       <tr key={rowIndex}>
+         {row.map((tile, tileIndex) => (
+           <td
```
Comment (jvalatkevicius) [inline services/app/src/components/chessBoard/ChessBoard.jsx:13]: Manau reiketu `Tile` iskelti i atkira komponenta ir padaryti taip
```
const ChessBoard = () => {
  const { placeItemOnBoard, board } = useContext(ChessBoardContext)

  return (
    <table id="chess-board">
      {board.map((row, rowIndex) => (
        <tr key={rowIndex}>
          {row.map((tile, columnIndex) => (
            <Tile
              key={tileIndex}
              attribute={tile.attribute}
              items={tile.items}
              onClick={placeItemOnBoard}
              row={rowIndex}
              column={columnIndex}
            />
          ))}
        </tr>
      ))}
    </table>
  );
};
```

Kur `Tile.jsx`:
```
const Tile = ({ attribute, items, onClick, row, column }) => {
  const someSensibleNameThatDescribesWhatThisConditionMeans = () => {
    return !tile.attribute && row !== board.length - 1 && column !== 0
  }

  const _onClick = useCallback(() => {
    onClick(row, column)
  }, [onClick]);

  return (
    <td
      onClick={_onClick}
      role='presentation'
      className={
        classNames({
          'black-tile': tile.isFilled,
          'white-tile': someSensibleNameThatDescribesWhatThisConditionMeans(),
        })
      }
    >
      {attribute || items}
    </td>
  )
}
```

cia classNames yra package'as, kuris is true/false objekto sulipdo klasiu vardus - https://www.npmjs.com/search?q=classNames

**Review by jvalatkevicius — CHANGES_REQUESTED** [review]

File: services/app/src/hooks/useChessBoard.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const alphabetArray = [...'abcdefghijklmnopqrstuvwxyz'];
+
+export const useChessBoard = () => {
+  const [board, setBoard] = useState(
+    [...Array(numberOfRows + 1).keys()].map((_row, rowIndex) =>
+      [...Array(numberOfColumns + 1).keys()].map((_tile, tileIndex) => ({
+        attribute:
+          tileIndex === 0 && rowIndex !== numberOfRows
+            ? numberOfColumns - rowIndex
```
Comment (raywiis) [inline services/app/src/hooks/useChessBoard.js:12]: why not get rid of one of the ternaries and explicitly have a row for numbers?
```
[
  ...Array(numberOfRows).keys().map(() => etc),
  [...[...Array(numberOfRows).keys()].map(k => k + 1))
]
```
Or just
```
[
  ...Board stuff,
  [1, 2, 3, 4, 5, 6, 7, 8]
]
```

Comment (raywiis) [reply services/app/src/hooks/useChessBoard.js:12]: Same with the symbol for columns
just add the symbol as an element before spreading the rest of the board stuff in and you won't need a 3 story ternary.

File: services/app/src/hooks/useChessBoard.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { useState } from 'react';
+import { numberOfRows, numberOfColumns } from '../constants/board';
+
+const alphabetArray = [...'abcdefghijklmnopqrstuvwxyz'];
+
+export const useChessBoard = () => {
+  const [board, setBoard] = useState(
+    [...Array(numberOfRows + 1).keys()].map((_row, rowIndex) =>
+      [...Array(numberOfColumns + 1).keys()].map((_tile, tileIndex) => ({
```
Comment (raywiis) [inline services/app/src/hooks/useChessBoard.js:9]: Python has a `range` function, you could make yourself something similar here, a function that returns an array from x to y

**Review by raywiis — CHANGES_REQUESTED** [review]

File: services/app/src/hooks/useChessBoard.js (hunk 3 above)
Comment (raywiis) [reply services/app/src/hooks/useChessBoard.js:12]: My bad: row for symbols, column for numbers. But the point is the same - have explicit elements to simplify the ternary.

**Review by raywiis — APPROVED** [review]

**Review by jvalatkevicius — APPROVED** [review]

**Comment by raywiis** [comment]
@tautvydasLiekis I think this needs your review

**Comment by albekruz** [comment]
> @tautvydasLiekis I think this needs your review

This ticket is still "in progress" as we decide to add some additional functionalities on the last catch-up meeting.

File: services/app/src/constants/board.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export const PIECES = ['pawn', 'knight', 'bishop', 'rook', 'king', 'queen'];
```
[the remark this answers is not in the document]
Comment (albekruz) [reply services/app/src/constants/board.js:1]: Kol kas neturim fronte .ts loaderio webpacke, tai reikėtu papildomai į webpacka instaliuot loaderius, kas nėra susiję su užduotim, tai ateityje jeigu nuspręsim įdėti .ts į frontą, tuomet bus galima pridėti ir enum'us.

File: services/api/src/game/game.service.ts [hunk 6]
```diff
@@ -1,7 +1,11 @@
-import logger from '../logger';
+const turns: any = [];
 
 export const figureMoved = (payload: unknown): any => {
-  //  TODO implement game logic
-  logger.info(`Figure moved to ${payload}`);
-  return {};
+  if (payload) {
+    turns.push(payload);
```
[the remark this answers is not in the document]
Comment (albekruz) [reply services/api/src/game/game.service.ts]: After clicking on submit button, player will not be able to update his figure position

File: services/api/src/domain/player.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
 export interface Player {
   id: string;
   name: string;
+  status: 'done' | 'active';
```
Comment (arvurb) [inline services/api/src/domain/player.ts:4]: use enum

**Review by arvurb — CHANGES_REQUESTED** [review]

**Review by arvurb — APPROVED** [review]

**Review by arvurb — DISMISSED** [review]

**Review by jvalatkevicius — APPROVED** [review]

File: services/app/src/components/chessBoard/ChessBoard.jsx [hunk 8]
```diff
@@ -1,54 +1,31 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+/* eslint-disable no-console */
```
Comment (jvalatkevicius) [inline services/app/src/components/chessBoard/ChessBoard.jsx:1]: Why is this here?

**Review by arvurb — APPROVED** [review]

**Review by jvalatkevicius — APPROVED** [review]

**Pull request merged**